File: mRemoteNG/Credential/Credential.List.vb
Namespace Credential
    Public Class List
        Inherits CollectionBase

#Region "Public Properties"
        Default Public ReadOnly Property Items(ByVal Index As Object) As Credential.Info
            Get
                If TypeOf Index Is Credential.Info Then
                    Return Index
                Else
                    Return CType(List.Item(Index), Credential.Info)
                End If
            End Get
        End Property

        Public Shadows ReadOnly Property Count() As Integer
            Get
                Return List.Count
            End Get
        End Property
#End Region

#Region "Public Methods"
        Public Function Add(ByVal cInfo As Credential.Info) As Credential.Info
            List.Add(cInfo)
            Return cInfo
        End Function

        Public Sub AddRange(ByVal cInfo() As Credential.Info)
            For Each cI As Credential.Info In cInfo
                List.Add(cI)
            Next
        End Sub

        Public Function Copy() As Credential.List
            Try
                Return Me.MemberwiseClone
            Catch ex As Exception
            End Try

            Return Nothing
        End Function

        Public Shadows Sub Clear()
            List.Clear()
        End Sub
#End Region
    End Class
End Namespace

File: mRemoteV1.ConvertedToC#/Credential/Credential.List.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
namespace mRemoteNG.Credential
{
	public class List : CollectionBase
	{

		#region "Public Properties"
		public Credential.Info this[object Index] {
			get {
				if (Index is Credential.Info) {
					return Index;
				} else {
					return (Credential.Info)List[Index];
				}
			}
		}

		public new int Count {
			get { return List.Count; }
		}
		#endregion

		#region "Public Methods"
		public Credential.Info Add(Credential.Info cInfo)
		{
			List.Add(cInfo);
			return cInfo;
		}

		public void AddRange(Credential.Info[] cInfo)
		{
			foreach (Credential.Info cI in cInfo) {
				List.Add(cI);
			}
		}

		public Credential.List Copy()
		{
			try {
				return this.MemberwiseClone();
			} catch (Exception ex) {
			}

			return null;
		}

		public new void Clear()
		{
			List.Clear();
		}
		#endregion
	}
}
